spec/emerald/compiler_spec.rb

```diff
@@ -17,4 +17,24 @@
       expect(compiled_lisp).to eq("")
     end
   end
+
+  context "when the contents of the file define a variable" do
+    it "parses the lisp and returns an assigned global variable" do
+      lisp_file_contents = "(def y (- 1 6))"
+      compiled_lisp = Emerald::Compiler.new(lisp_file_contents).compile
+      expect(compiled_lisp).to eq("y = 1.0 - 6.0")
+    end
+
+    it "parses the lisp and returns an assigned local variable" do
+      lisp_file_contents = "(let [x 1 y (+ 3 4)] x)"
```

Comment:
Oh, now I remember, it was because I was having difficulty parsing local variables if they were defined with `(...)`. My current method of parsing matches start and end brackets, and includes everything inside (including other lists). So say if we had this local variable definition:
```lisp
(let ((x 1)) (+ x 3 ))
```
When it comes to parsing two adjacent lists, this:
```
((x 1)) (+ x 3 )
```
would parse as:
```lisp
(x 1)) (+ x 3 
```
I considered other methods of parsing lists e.g. by tracking the number of open/closed brackets, but then that got messy when I tried to account for brackets within quotes. Do you have any suggestions? :)